pkg/webui/lib/selectors/profile-picture.js

```diff
@@ -0,0 +1,71 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export const isValidProfilePictureObject = profilePicture =>
+  Boolean(profilePicture) && Boolean(profilePicture.sizes)
+
+export const getProfilePictureSizes = profilePicture => {
+  if (!isValidProfilePictureObject(profilePicture))
+    throw new Error('Invalid profile picture object')
+  return profilePicture.sizes
+}
+
+export const getLargestAvailableProfilePicture = profilePicture => {
+  const sizes = getProfilePictureSizes(profilePicture)
+  const sizeValues = Object.keys(profilePicture.sizes)
+
+  return sizes[0] || sizes[Math.max(...sizeValues)]
+}
+
+export const getSmallestAvailableProfilePicture = profilePicture =>
+  getClosestProfilePictureBySize(profilePicture, 64)
+
+export const getOriginalSizeProfilePicture = profilePicture => {
+  const sizes = getProfilePictureSizes(profilePicture)
+
+  return sizes[0]
+}
+
+export const getClosestProfilePictureBySize = (profilePicture, size) => {
```

Comment:
Closest to `> 'size'` or `< 'size'` ?

Reply:
I've updated the function, it will return whatever is closer to the requested size. Since the PPs use powers of two, I didn't include any logic for the case that there are two sizes with the same distance, since it should not occur.